File: src/utils/fetchYouTubeVideos.py
import requests
import xml.etree.ElementTree as ET

CHANNEL_ID = 'UC-db2t_fVN3T_kzKYSNUioA'
YOUTUBE_CHANNEL = 'https://www.youtube.com/feeds/videos.xml?channel_id={CHANNEL_ID}'.format(CHANNEL_ID=CHANNEL_ID)

response = requests.get(YOUTUBE_CHANNEL)

root = ET.fromstring(response.content)

videos = []
firstTwoVideos = []
for element in root.iter('*'):
    if 'href' in element.attrib:
        videos.append(element.attrib['href'].replace('watch?v=', 'embed/'))

firstTwoVideos.append(videos[2])
firstTwoVideos.append(videos[3])


videosFile = open('./src/utils/constants/videos.js', 'w')
videosFile.write('export default {firstTwoVideos} // hihi'.format(firstTwoVideos=firstTwoVideos))
videosFile.close()

File: src/utils/fetchYouTubeVideos.js
const CORS_PROXY = 'https://cors-anywhere.herokuapp.com/'
const CHANNEL_ID = 'UC-db2t_fVN3T_kzKYSNUioA';
const YOUTUBE_CHANNEL = `${CORS_PROXY}https://www.youtube.com/feeds/videos.xml?channel_id=${CHANNEL_ID}`;

export default (setYouTubeVideos) => {
  fetch(YOUTUBE_CHANNEL).then(async (res) => {
    const youTubeVideosData = await res.text();
    const parser = new DOMParser();
    const xmlDoc = parser.parseFromString(youTubeVideosData, 'text/xml');
    const allLinks = Array.from(xmlDoc.querySelectorAll('link'));
    const firstTwoVideos = allLinks.reduce((links, currentLink, index) => {
      if (index === 2 || index === 3) {
        links.push(currentLink.getAttribute('href').replace('watch?v=', 'embed/'))
      }
      return links;
    }, []);
    setYouTubeVideos(firstTwoVideos);
  });
}
